src/ucp/core/ucp_context.c

```diff
@@ -1367,6 +1391,13 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
      * routines */
     UCP_THREAD_LOCK_INIT(&context->mt_lock);
 
+    /* save environment prefix to later notify user for unused variables */
+    context->config.env_prefix = ucs_strdup(config->env_prefix, "ucp config");
+    if (context->config.env_prefix == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto err_strdup;
```

Comment:
no need to `goto err_strdup`, can just goto `err` (even thought `free(NULL)` is ok)
but you need to `goto err_strdup` below on lines 1405 and 1417